drivers/overlay/overlay.go

```diff
@@ -188,6 +188,9 @@ func (d *driver) configure() error {
 		return d.initializeVxlanIdm()
 	}
 
+	// Apply OS specific kernel configs if needed
+	d.once.Do(applyOStweaks)
+
```

Comment:
nit: can you change the `once` variable to something specific, maybe `osInitOnce` to avoid this getting inadvertently used for some other purpose during the driver init.

LGTM otherwise.